### ISSUE TITLE
Jasper Report: `Account Sheet`

### ISSUE THREAD

**Issue by TheBestPessimist** [metasfresh/metasfresh#6121]
### Is this a bug or feature request?
FR

**Comment by TheBestPessimist** [comment]
# Trying to figure out why `Account Sheet` returns different balance compared to `Saldobilanz`



`Saldobilanz` uses the sql function `acctBalanceToDate` to compute the balance, so all results of `Account Sheet` should match those.

### Saldobilanz report with date 2020-02-01

Amount is `-3'344'306.88`.

Same result as
```sql
SELECT *
FROM
    de_metas_acct.acctBalanceToDate(1002995,
                                    1000000,
                                    '2020-02-01'::date,
                                    1000000,
                                    'N',
                                    'N');
```




### Saldobilanz report with date 2020-02-02

Amount is `-3'353'615.50`

Same result as
```sql
SELECT *
FROM
    de_metas_acct.acctBalanceToDate(1002995,
                                    1000000,
                                    '2020-02-02'::date,
                                    1000000,
                                    'N',
                                    'N');
```


### Account Sheet for period 2020-02-01 - 2020-02-02


![image](https://user-images.githubusercontent.com/4482210/76495610-e53e3d00-643f-11ea-9bf4-c3d79753e011.png)


It seems that `Account sheet` is adding that credit (haben) to the beginning value. If it would subtract the credit, then the ending balances would match between the 2 reports.

-3342603.64 - 283.05 - 1308.39 - 111.8 = -3,353,615.5


The problem seems to be in the following sql in the `Account Sheet` report:
```sql
    --
    -- Update the current balance for each row.
    -- This implementation uses a rolling sum over the previous rows
    WITH beginningBalance_fa AS
             (
                 SELECT tmp_fa.fact_acct_id,
                        (
                                tmp_fa.endingBalance
                                + sum(acctbalance(tmp_fa.account_id, tmp_fa.amtacctdr, tmp_fa.amtacctcr)) -- !!!! acctbalance should return negative here
                                  OVER
                                      (
                                      PARTITION BY tmp_fa.account_id
                                      ORDER BY tmp_fa.dateacct, tmp_fa.fact_acct_id ROWS BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW
                                      )
                            ) AS                                                           beginningBalance,
                        acctbalance(tmp_fa.account_id, tmp_fa.amtacctdr, tmp_fa.amtacctcr) transactionBalance
                 FROM TMP_AccountSheetReport tmp_fa
             ),
```

It seems `acctbalance(tmp_fa.account_id, tmp_fa.amtacctdr, tmp_fa.amtacctcr)` should return negative, but it returns a positive value.

In `acctBalance`, it could have something to do with  `c_elementValue.AccountSign` and `AccountType`.

Another detail is that `acctBalanceToDate` (the "norm") doesn't use `acctBalance` function while `Account Sheet` uses it, so it may very well be that `acctBalanceToDate` is returning wrong results (though i find that hard to believe, as it was used very much so far and nobody complained).

I would like to understand what are  `c_elementValue.AccountSign` and `AccountType` and how should they be used, but for this i need either @teosarca or @metas-mk 's help.

---

# Solution after talking with @metas-mk.

I talked with mark and he explained that the solution is to use `debit - credit`, instead of the function `acctBalance`. This is required because all other calculation (`Saldobilanz` which uses the function `acctBalanceToDate`) do not use `acctBalance` either. 
If they were all using `acctBalance` instead of `debit - credit`, we would have correct answers. 

Basically this is not a problem of correctness in the absolute sense, it's a problem of consistency. The user is used to having an easy calculation of `debit - credit` in this case, so we shouldn't change his habbit but adapt.
